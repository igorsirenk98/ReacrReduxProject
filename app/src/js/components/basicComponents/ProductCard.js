import React from 'react';
import { Link } from 'react-router-dom';

import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';

const listItemStyles = {
    margin: '0 auto',
    width: 'auto'
};

const cardStyles = {
    'margin': '20px'
};

const linkStyles = {
    'textDecoration': 'none'
};


export const ProductCard = props => {
    console.log(props);
    const { product } = props;

    return (
        <>
            {product && 
                <Card style={cardStyles}>
                    <CardActionArea style={listItemStyles}>
                        <img className='MuiCardMedia-root' src={`data:image/jpeg;base64, ${product['productProductPhoto.productPhoto.largePhoto']}`}/>
                        <CardContent>
                            <Typography gutterBottom variant="h5" component="h2">
                                {product.name}
                            </Typography>
                            <Typography variant="body2" color="textSecondary" component="p">
                                Type: {product['productSubcategory.name']}
                            </Typography>
                            <Typography variant="body2" color="textSecondary" component="p">
                                Price: {product.listPrice}$
                            </Typography>
                        </CardContent>
                    </CardActionArea>
                    {!product['productDescriptionCulture.productDescription.description'] &&
                        <CardActions>
                            <Button size="small" color="primary">
                                <Link
                                    style={linkStyles}
                                    key={product.productId}
                                    to={{
                                        pathname: `/products/productId=${product.productId}`,
                                        productId: product.productId
                                    }}
                                >
                                    Details
                                </Link>
                            </Button>
                        </CardActions>}
                </Card>
            }
        </>
    );
}