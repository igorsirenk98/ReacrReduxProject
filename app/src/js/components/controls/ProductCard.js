import React from 'react';
import { Link } from 'react-router-dom';

import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import Tooltip from '@material-ui/core/Tooltip';
import Zoom from '@material-ui/core/Zoom';

const ProductCard = (props) => {
    const { product } = props;

    return (
        <>
            <Link
                className="products__item link"
                key={product.productId}
                to={{
                    pathname: `/products/productId=${product.productId}`,
                    productId: product.productId,
                }}
            >
                <Tooltip title="Click to view details" TransitionComponent={Zoom} arrow>
                    <Card className="card">
                        <CardActionArea className="products__item">
                            <img
                                className="img"
                                src={`data:image/jpeg;base64, ${product.largePhoto}`}
                                alt={product.name}
                            />
                            <CardContent>
                                <Typography gutterBottom variant="h5" component="h2">
                                    {product.name}
                                </Typography>
                                <Typography variant="body2" color="textSecondary" component="p">
                                    Type: {product['productSubcategory.name']}
                                </Typography>
                                <Typography variant="body2" color="textSecondary" component="p">
                                    Price: {product.listPrice}$
                                </Typography>
                            </CardContent>
                        </CardActionArea>
                    </Card>
                </Tooltip>
            </Link>
        </>
    );
};

export default ProductCard;
