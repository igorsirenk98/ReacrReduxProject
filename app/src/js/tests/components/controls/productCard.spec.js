import React from 'react'
import { shallow } from 'enzyme'
import ProductCard from '../../../components/controls/ProductCard';

describe('ProductCard component', () => {
    const productCard = shallow(<ProductCard product={{
        productId: 1,
        name: 'bike',
        productSubcategory: {
            name: 'road'
        },
        listPrice: 100
    }} />);

    it('renders correctly', () => {
        expect(productCard.debug()).toMatchSnapshot()
    })
})
