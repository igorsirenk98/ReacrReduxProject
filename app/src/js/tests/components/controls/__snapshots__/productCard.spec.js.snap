// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component renders correctly 1`] = `
"<Fragment>
  <Link className=\\"products__item link\\" to={{...}}>
    <WithStyles(ForwardRef(Tooltip)) title=\\"Click to view details\\" TransitionComponent={{...}} arrow={true}>
      <WithStyles(ForwardRef(Card)) className=\\"card\\">
        <WithStyles(ForwardRef(CardActionArea)) className=\\"products__item\\">
          <img className=\\"img\\" src=\\"data:image/jpeg;base64, undefined\\" alt=\\"bike\\" />
          <WithStyles(ForwardRef(CardContent))>
            <WithStyles(ForwardRef(Typography)) gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
              bike
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" color=\\"textSecondary\\" component=\\"p\\">
              Type: 
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" color=\\"textSecondary\\" component=\\"p\\">
              Price: 
              100
              $
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(CardContent))>
        </WithStyles(ForwardRef(CardActionArea))>
      </WithStyles(ForwardRef(Card))>
    </WithStyles(ForwardRef(Tooltip))>
  </Link>
</Fragment>"
`;
